feeds/npm/npm.go

```diff
@@ -75,19 +50,49 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
-// Gets the package version from the NPM.
-func fetchVersionInformation(packageName string) (string, error) {
-	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, packageName))
+// Gets the package version & corresponding created date from NPM.
+func fetchVersionInformation(pkg *Package) (*Package, error) {
+	resp, err := httpClient.Get(fmt.Sprintf("%s/%s", versionURL, pkg.Title))
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	defer resp.Body.Close()
-	v := &PackageVersion{}
-	err = json.NewDecoder(resp.Body).Decode(v)
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	var jsonMap map[string]interface{}
+	err = json.Unmarshal(body, &jsonMap)
+	if err != nil {
+		return nil, err
+	}
+	distTags, ok := jsonMap["dist-tags"].(map[string]interface{})
+	if !ok {
+		return nil, errJSON
+	}
+	versions, ok := jsonMap["time"].(map[string]interface{})
+	if !ok {
+		return nil, errJSON
+	}
+
+	for _, v := range distTags {
+		versionString, ok := v.(string)
+		if !ok {
+			return nil, errJSON
+		}
+		date, err := time.Parse(time.RFC3339, versions[versionString].(string))
+		if err != nil {
+			return nil, err
+		}
+		if pkg.CreatedDate.IsZero() {
```

Comment:
Can these two statements be merged to remove the duplication?

Reply:
I'm not sure what you mean here, what is checking if `time.Parse()` is returning an error on the json string, `pkg.CreatedDate.IsZero()` is checking if the pkg struct has a zero value for the field, in this case meaning the first iteration.

Reply:
`.IsZero` and `.After` both do the same logic in the case of True, these can be anded together.

Reply:
Ah, I think I see what you mean, although I think in this case it's an `or`